File: unit-2/sprint-2/day-2-RecyclerView/assignments/Article Scroll/app/src/main/java/com/example/article/MainActivity.java
package com.example.article;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements ItemClickListener{

    private RecyclerView recyclerView;
    private ArrayList<Model> modelList;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = findViewById(R.id.rvRecyclerView);
        buildData();
        setRecyclerAdapter();
    }
    private void buildData() {
        modelList = new ArrayList<>();
        for (int i = 0; i < 100; i++) {
            if (i % 2 == 0)
                modelList.add(new Model(R.drawable.th_1,"Blue Ocean Wave Crashed","As waves crashed against the shoreline in Encinitas on Monday, a bright blue hue overtook the ocean thanks to the return of an unpredictable natural phenomenon.The bright blue light, created by phytoplankton through a process called bioluminescence.","Mark"));
            else if (i % 2 == 1)
                modelList.add(new Model(R.drawable.th_1,"Blue Ocean Wave Crashed","As waves crashed against the shoreline in Encinitas on Monday, a bright blue hue overtook the ocean thanks to the return of an unpredictable natural phenomenon.The bright blue light, created by phytoplankton through a process called bioluminescence.","Mark"));
        }
    }
    private void setRecyclerAdapter(){
        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        ModelAdaptor modelAdaptor = new ModelAdaptor(modelList,this);
        recyclerView.setAdapter(modelAdaptor);
    }

    @Override
    public void onItemClicked(Model model) {
        Toast.makeText(this,model.getAuthor(),Toast.LENGTH_SHORT).show();
    }
}

File: unit-5/sprint-4(Coding)/app/src/main/java/com/example/my/ui/MainActivity.kt
package com.example.my.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.my.R
import com.example.my.data.models.Article
import com.example.my.ui.adapter.MainAdapter
import com.example.my.ui.viewmodel.MainViewModel
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private lateinit var mainViewModel : MainViewModel
    private lateinit var mainAdapter: MainAdapter
    var list = ArrayList<Article>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mainViewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        setRecyclerAdapter()
        mainViewModel.callAPI()

        mainViewModel.liveData.observe(this,{
            list.clear()
            list.addAll(it)
            mainAdapter.notifyDataSetChanged()
        })
    }

    private fun setRecyclerAdapter() {
        mainAdapter = MainAdapter(list)
        val linearLayoutManager = LinearLayoutManager(this)
        recyclerView.apply {
            adapter = mainAdapter
            layoutManager = linearLayoutManager
        }
    }
}